### PR TITLE
feat: apps deploy improvements

### PR DESCRIPTION
**Pull request by raulb**
## Description of change

Fixes https://github.com/meroxa/turbine-project/issues/132

- It also makes some changes on some other build commands.
- It avoids printing out an empty line when there's nothing to log out: https://github.com/meroxa/cli/pull/301/commits/488cf68e3ed007e6df0ae362f8748be79db69644
- It shows a friendlier message when there's an invalid token: https://github.com/meroxa/cli/pull/301/commits/1c8cc6b584ce5feea19761f8a6bb68dc98003d05
- It adds spinner through the deployment process
- It adds a way to include spinners across the code as part of logger
- It updates the deployment output

## Type of change

- [x]  New feature
- [ ]  Bug fix
- [x]  Refactor
- [ ]  Documentation

## How was this tested?

- [ ]  Unit Tests
- [ ]  Tested in staging
- [ ]  Tested in minikube

## Demo

See slack! I recorded a couple.

### CODE REVIEW

File: cmd/meroxa/turbine_cli/utils.go [hunk 1]
```diff
@@ -332,6 +332,9 @@ func RunCmdWithErrorDetection(ctx context.Context, cmd *exec.Cmd, l log.Logger)
 		return "", errors.New(errLog)
 	}
 	l.Info(ctx, stdOutMsg)
```
Comment (raulb) [inline cmd/meroxa/turbine_cli/utils.go]: Fixed by https://github.com/meroxa/cli/pull/301/commits/d399bd2a5ac4afe60adade5a2249dc91bd0a867a 😅

File: cmd/meroxa/root/apps/deploy.go [hunk 2]
```diff
@@ -203,14 +205,19 @@ func (d *Deploy) createApplication(ctx context.Context, pipelineUUID, gitSha str
 				return err
 			}
 			if app.Pipeline.UUID.String != pipelineUUID {
-				return fmt.Errorf("unable to finish creating the %s Application because its entities are in an"+
-					" unrecoverable state; try deleting and re-deploying", appName)
+				d.logger.StopSpinner(fmt.Sprintf("\t  𐄂 unable to finish creating the %s Application because its entities are in an"+
+					" unrecoverable state; try deleting and re-deploying", appName))
+
+				// TODO: Rollback here?
```
Comment (janelletavares) [inline cmd/meroxa/root/apps/deploy.go:211]: oh, so rollback is a feature of the CLI, not of the language specific libraries? or both?

Comment (raulb) [reply cmd/meroxa/root/apps/deploy.go:212]: @janelletavares it should be library specific I think

**Comment by janelletavares** [comment]
Screenshot from 2022-04-07 18-14-17

the something-bad-happened emojis don't look like anything in linux

**Review by janelletavares — APPROVED** [review]

**Comment by janelletavares** [comment]
@raulb I'm finished reviewing and testing.  Let me know if you want to discuss anything that I pushed in my commits.

File: cmd/meroxa/turbine_cli/utils.go [hunk 3]
```diff
@@ -353,7 +356,7 @@ func CreateTarAndZipFile(src string, buf io.Writer) error {
 	zipWriter := gzip.NewWriter(buf)
 	tarWriter := tar.NewWriter(zipWriter)
 
-	err = filepath.Walk(src, func(file string, fi os.FileInfo, err error) error {
+	err = filepath.Walk(appDir, func(file string, fi os.FileInfo, err error) error {
```
Comment (raulb) [inline cmd/meroxa/turbine_cli/utils.go:359]: This broke for me the other day. I'll check if it still works

Comment (raulb) [reply cmd/meroxa/turbine_cli/utils.go:359]: Just checked. It still works! ❇️

**Review by jmar910 — APPROVED** [review]

**Review by jayjayjpg — COMMENTED** [review]
This reads great! I tested this in combination with the turbine-js branch `james/fix-deploy-past-n-future` and this seems to return an error when using the `meroxa apps deploy` command when deploying a JS app

I'm unsure if this is related to the Turbine lib or the CLI changes though, what do you think?

### Deploy with --path

initappsjs

### Deploy from current directory

initappsjs2

initappsjs3

**Review by jayjayjpg — APPROVED** [review]
I just noticed after looking into this more that this is working as expected and that app deployments only fail under certain circumstances unrelated to this change

This reads great and works well, thank you for the super swift fix! :sparkles:

jsinitagain

Just a heads up, that I tracked the unrelated issue as a bug ticket [here](https://github.com/meroxa/turbine-project/issues/145)

**Pull request merged**